src/lib/ThumbnailsSidebar.js

```diff
@@ -212,19 +228,29 @@ class ThumbnailsSidebar {
      * Make a thumbnail image element
      *
      * @param {number} itemIndex - the item index for the overall list (0 indexed)
-     * @return {Promise} - promise reolves with the image HTMLElement
+     * @return {Promise} - promise reolves with the image HTMLElement or null if generation is in progress
      */
     createThumbnailImage(itemIndex) {
-        // If this page has already been cached, use it
-        if (this.thumbnailImageCache[itemIndex]) {
-            return Promise.resolve(this.thumbnailImageCache[itemIndex]);
+        const cacheEntry = this.thumbnailImageCache[itemIndex];
+
+        // If this thumbnail has already been cached, use it
+        if (cacheEntry && cacheEntry.image) {
```

Comment:
It might be useful in the future to have a cache class with utility methods

Reply:
Sure I'll put that on my thumbnails todo list